### PR TITLE
Replace Install Juju button with sandbox/demo notification

### PR DESCRIPTION
**Pull request by anthonydillon**
A prominent notification at the bottom of the screen to inform your in sandbox mode.

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Open the GUI locally and you should see a dark bar along the bottom stating its a demo
- Check that the Install Juju button has been removed
- Check that the deployment bar is moved up when in sandbox to make room for the notification
- Click the Get started button and it should open the get-started page in a new tab
- Set the GUI to none sandbox or run it in a real environment and check the bar is not there
- Check the deployment bar is back to its original position

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4521/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4522/
Test PASSed.

File: jujugui/static/gui/src/app/components/demo-notification/demo-notification.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      sandboxMode: React.PropTypes.bool.isRequired
+    },
+
+    /**
+      Get the current state of the demo notification.
+
+      @method getInitialState
+      @returns {Object} The current state.
+    */
+    getInitialState: function() {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/demo-notification/demo-notification.js]: This function is not necessary you don't have any component state

File: jujugui/static/gui/src/app/components/demo-notification/demo-notification.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('demo-notification', function() {
+
+  juju.components.DemoNotification = React.createClass({
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/demo-notification/demo-notification.js]: This component looks like a good case for https://facebook.github.io/react/docs/reusable-components.html#stateless-functions

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4523/
Test PASSed.

File: jujugui/static/gui/src/app/components/demo-notification/test-demo-notification.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    // By loading this file it adds the component to the juju components.
+    YUI().use('demo-notification', function() { done(); });
+  });
+
+  beforeEach(function() {
+    var DemoNotification = juju.components.DemoNotification;
+    DemoNotification.prototype.sandboxMode = true;
+  });
+
+  xit('can render and in sandbox mode', function() {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/demo-notification/test-demo-notification.js]: This test is currently skipped...

**Comment by huwshimi** [comment]
Thanks looks good @anthonydillon. The message appeared in sandbox mode and was hidden in a real env.

One QA issue: notifications appear under the message, I guess they'll need to be moved up too.

:+1: with the QA and test issue fixed.

**Comment by anthonydillon** [comment]
Ah good spot with the notifications. I should have mentioned but not sure shallow renderer works with stateless functions so skipped it cos I didn't want the test file to be empty. Also there is no functionality or state so don't think the notification needs tests. What do you think? Can the test file be removed?

**Comment by huwshimi** [comment]
@anthonydillon From memory you don't have to use the shallow renderer, you can just call the stateless function directly to get the output, see: https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/components/deployment-summary/test-deployment-summary-placement.js

As for whether or not it needs a test, I'm not sure. It might be nice to have a render test in case we add functionality in the future and then we don't have to do the test setup and we know if we've broken something. @hatched do you have any opinions on this?

**Comment by hatched** [comment]
@huwshimi That sounds correct I believe @anthonydillon and I talked about that at EOD on Friday.

**Comment by anthonydillon** [comment]
This is being closed as discussed in the issue to go no further with this.

**Pull request closed without merging**